src/app/shared/context-help.service.ts

```diff
@@ -0,0 +1,113 @@
+import { Injectable } from '@angular/core';
+import { ContextHelp } from './context-help.model';
+import { Store, createFeatureSelector, createSelector, select, MemoizedSelector } from '@ngrx/store';
+import { ContextHelpState, ContextHelpModels } from './context-help.reducer';
+import {
+  ContextHelpToggleIconsAction,
+  ContextHelpAddAction,
+  ContextHelpRemoveAction,
+  ContextHelpShowTooltipAction,
+  ContextHelpHideTooltipAction,
+  ContextHelpToggleTooltipAction
+} from './context-help.actions';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+const contextHelpStateSelector =
+  createFeatureSelector<ContextHelpState>('contextHelp');
+const allIconsVisibleSelector = createSelector(
+  contextHelpStateSelector,
+  (state: ContextHelpState): boolean => state.allIconsVisible
+);
+const contextHelpSelector =
+  (id: string): MemoizedSelector<ContextHelpState, ContextHelp> => createSelector(
+    contextHelpStateSelector,
+    (state: ContextHelpState) => state.models[id]
+  );
+const allContextHelpSelector = createSelector(
+  contextHelpStateSelector,
+  ((state: ContextHelpState) => state.models)
+);
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ContextHelpService {
+  constructor(private store: Store<ContextHelpState>) { }
+
+  /**
+   * Observable keeping track of whether context help icons should be visible globally.
+   */
+  shouldShowIcons$(): Observable<boolean> {
```

Comment:
the $ suffix is usually used for variables not for the methods. I'd rename all the methods with $ suffix

Reply:
I'm not sure I agree with that. Sure it's not the current convention in DSpace, but adding the dollar suffix makes it very clear that it'll return an observable, and I can't imagine what the downside would be

Reply:
My two cents: Adding the suffix to method names makes naming the sample in lambdas easier. For instance, if we write `shouldShowIcons$().subscribe(shouldShowIcons -> ...)`, we can have both the observable(-producing method) and its sample in scope without issue. Omitting the $ suffix means we have to either shadow the outer variable (disallowed by the linter, IIRC), or come up with some name variation for the sample every time, neither of which seem satisfying.

That said, I also follow the argument for consistency with other method names, and I'll remove the suffix if you prefer it.

Reply:
We appear to already have many methods that end in a dollar sign ($).  A quick search of the codebase shows we have methods ending in a dollar sign in `theme.service.ts`, `init.service.ts`, `menu.resolver.ts`, etc.  In all instances, it looks like those methods are returning Observables, just like this one.

We also have examples of many methods (sometimes even in the same file, see `menu.resolver.ts`) that return Observables but do *not* end in a dollar sign.

So, this is just to point out that we don't seem to have any firm agreement as to whether methods can or cannot end in a dollar sign.  Therefore, this code is fine as-is, or you can change it. It's really up to @KoenP and this request should be considered *optional* as we don't have a firm best practice.

(If we wish to establish a best practice in the future, we can always do so. But, at this point, we just don't have one.)

Reply:
I've left this as-is for now.